crates/router/src/core/admin.rs

```diff
@@ -874,3 +871,210 @@ pub fn get_frm_config_as_secret(
         None => None,
     }
 }
+
+pub(crate) fn validate_auth_type(
+    connector_name: api_models::enums::Connector,
+    val: &types::ConnectorAuthType,
+) -> Result<(), error_stack::Report<errors::ConnectorError>> {
+    use crate::connector::*;
+
+    match connector_name {
+        api_enums::Connector::DummyConnector1 => {
+            dummyconnector::transformers::DummyConnectorAuthType::try_from(val)?;
+            Ok(())
+        }
+        api_enums::Connector::DummyConnector2 => {
+            dummyconnector::transformers::DummyConnectorAuthType::try_from(val)?;
+            Ok(())
+        }
+        api_enums::Connector::DummyConnector3 => {
+            dummyconnector::transformers::DummyConnectorAuthType::try_from(val)?;
+            Ok(())
+        }
+        api_enums::Connector::DummyConnector4 => {
+            dummyconnector::transformers::DummyConnectorAuthType::try_from(val)?;
+            Ok(())
+        }
+        api_enums::Connector::DummyConnector5 => {
+            dummyconnector::transformers::DummyConnectorAuthType::try_from(val)?;
+            Ok(())
+        }
+        api_enums::Connector::DummyConnector6 => {
+            dummyconnector::transformers::DummyConnectorAuthType::try_from(val)?;
+            Ok(())
+        }
+        api_enums::Connector::DummyConnector7 => {
+            dummyconnector::transformers::DummyConnectorAuthType::try_from(val)?;
+            Ok(())
+        }
+        api_enums::Connector::Aci => {
+            aci::transformers::AciAuthType::try_from(val)?;
+            Ok(())
+        }
+        api_enums::Connector::Adyen => {
+            adyen::transformers::AdyenAuthType::try_from(val)?;
+            Ok(())
+        }
+        api_enums::Connector::Airwallex => {
+            airwallex::transformers::AirwallexAuthType::try_from(val)?;
+            Ok(())
+        }
+        api_enums::Connector::Authorizedotnet => {
+            authorizedotnet::transformers::MerchantAuthentication::try_from(val)?;
```

Comment:
@Narayanbhat166 Should we consider naming the type as `AuthorizedotnetAuthType` instead?